### ISSUE TITLE
Local backup / restore from file not working?

### ISSUE THREAD

**Issue by frobijn** [osmandapp/OsmAnd#12419]
### Description
Great feature, local backup! But the "restore" does not seem to work??? Or I may not understand how restore works.

I've created a local file backup (only favourites in My Places, not sure whether other settings are supposed to be bacup up). I've changed the content of the *.obf file (replaced the favourites) and selected the modified file for restore. The list of favourites in OSMAnd does not change.

To verify that the updated favourites.gpx is valid, I've deleted all favourites in OsmAnd and imported the favourites.gpx file. That worked fine: all locations are imported correctly.

### How to reproduce?
1. Start with OSMAnd that has some favourites
2. Select Backup to file..,
3. Rename the *.osf file to *.osf.zip
4. Replace the favourites.gpx in the zip file: [Export_08-08-21.osf.zip]https://github.com/osmandapp/OsmAnd/files/6951459/Export_08-08-21.osf.zip)
5. Rename the *.osf.zip file to *.osf
6. Select Restore from file... and select the updated *.osf

The list of favourites in My Places in OSMand is unmodified.

### Your Environment
OsmAnd Version: 4.0.4
Android/iOS version: 11
Device model: Samsung Galaxy S10+

**Comment by Chumva** [comment]
@frobijn After you selected this osf file, did a screen appear with a choice of what to restore from it?
If not, I suppose there were something wrong with your zip file, I have repacked it and restore worked. 

![Screenshot_1628700124](https://user-images.githubusercontent.com/19220417/129069099-4494b82e-e151-4b13-9b9e-42de6daafd71.png)

https://drive.google.com/file/d/1GXX2mzqDAWVq7eKZtJVMBBDptWMdNdPM/view?usp=sharing

**Comment by vshcherb** [comment]
After some tries, we couldn't reproduce that issue so far.

**Comment by frobijn** [comment]
Indeed this is a weird issue.

I still cannot restore the *.osf file that I've created in the way I described. I don't get the "Import" option. But if I re-create the file the import succeeds.

I've examined the *.osf/*.zip file further. Windows can open it as a zip file, but cannot update the zip file. The file was created by updating the *.osf.zip file with IZArc; apparently that corrupted the file in some way.

Sorry to have waisted your time.
